### PR TITLE
Feature/442 Cloud Messaging

### PR DESCRIPTION
**Pull request by friedger**
This PR uses Google Service plugin 3.0 and the firebase configuration for FCM.

This is the preparation for more messaging logic. For now, we can't send messages to users per chapter, just the usual when in background messages. More depends on the backend.

screenshot_20160607-105354

### CODE REVIEW

**Comment by friedger** [comment]
See #442

File: app/google-services.json [hunk 1]
```diff
@@ -64,14 +62,7 @@
       ],
       "services": {
         "analytics_service": {
-          "status": 2,
-          "analytics_property": {
-            "tracking_id": "UA-42015512-1"
```
Comment (tasomaniac) [inline app/google-services.json]: I am afraid about these removal of analytics tracking ids. Is this json file generated by firebase?

File: app/src/main/res/values/tracker.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <string name="ga_trackingId">UA-42015512-1</string>
```
Comment (tasomaniac) [inline app/src/main/res/values/tracker.xml]: We had this implementation before. I don't know what is the best. I think if we use google's plugin, we should also use it for analytics. What do you think?

File: app/google-services.json (hunk 1 above)
Comment (friedger) [inline app/google-services.json]: It is generated by firebase. This is the solution suggested here http://stackoverflow.com/a/37549781/243599 to use Firebase alongside GA.

File: app/src/main/res/values/tracker.xml (hunk 2 above)
Comment (friedger) [inline app/src/main/res/values/tracker.xml]: I didn't want to switch the analytics for this PR.

Comment (tasomaniac) [inline app/src/main/res/values/tracker.xml]: I understand the reason now. Thank you for the stackoverflow link.

**Comment by tasomaniac** [comment]
:+1:

**Comment by tasomaniac** [comment]
:+1:

**Pull request merged**